jerry-core/ecma/base/ecma-helpers.h

```diff
@@ -52,6 +52,33 @@
  */
 #define ECMA_SET_POINTER(field, non_compressed_pointer) MEM_CP_SET_POINTER (field, non_compressed_pointer)
 
+/**
+ * Convert ecma-string's contents to a cesu-8 string and put it into a buffer.
+ */
+#define ECMA_STRING_TO_UTF8_STRING(ecma_str_ptr, utf8_ptr) \
+  lit_utf8_size_t utf8_ptr ## _size; \
+  bool utf8_ptr ## _is_ascii; \
+  const lit_utf8_byte_t *utf8_ptr = ecma_string_raw_chars (ecma_str_ptr, &utf8_ptr ## _size, &utf8_ptr ## _is_ascii); \
+  bool utf8_ptr ## must_be_freed = false; \
+  \
+  if (utf8_ptr == NULL) \
+  { \
+    utf8_ptr = (const lit_utf8_byte_t *) (mem_heap_alloc_block (utf8_ptr ## _size)); \
+    lit_utf8_size_t sz = ecma_string_to_utf8_string (ecma_str_ptr, (lit_utf8_byte_t *) utf8_ptr, utf8_ptr ## _size); \
+    JERRY_ASSERT (sz == utf8_ptr ## _size); \
+    utf8_ptr ## must_be_freed = true; \
+  }
```

Comment:
One more thing: it is intentional to pass back the size of the string by ecma_string_raw_chars as well. This should make string size getters redundant.